{
  "url": "https://api.github.com/repos/facebook/react/issues/14436",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/14436/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/14436/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/14436/events",
  "html_url": "https://github.com/facebook/react/issues/14436",
  "id": 390808310,
  "node_id": "MDU6SXNzdWUzOTA4MDgzMTA=",
  "number": 14436,
  "title": "linting with react hooks does not error when putting useState in a component",
  "user": {
    "login": "pkellner",
    "id": 241170,
    "node_id": "MDQ6VXNlcjI0MTE3MA==",
    "avatar_url": "https://avatars0.githubusercontent.com/u/241170?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/pkellner",
    "html_url": "https://github.com/pkellner",
    "followers_url": "https://api.github.com/users/pkellner/followers",
    "following_url": "https://api.github.com/users/pkellner/following{/other_user}",
    "gists_url": "https://api.github.com/users/pkellner/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/pkellner/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/pkellner/subscriptions",
    "organizations_url": "https://api.github.com/users/pkellner/orgs",
    "repos_url": "https://api.github.com/users/pkellner/repos",
    "events_url": "https://api.github.com/users/pkellner/events{/privacy}",
    "received_events_url": "https://api.github.com/users/pkellner/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 1,
  "created_at": "2018-12-13T18:40:49Z",
  "updated_at": "2019-01-18T03:11:48Z",
  "closed_at": "2019-01-18T03:11:48Z",
  "author_association": "NONE",
  "body": "I would have expected that calling a hook inside a react component would cause a linting error. From the docs it says \"Only Call Hooks from React Functions\"\r\n\r\n```import React, {useState} from 'react';\r\nclass axioshooks extends Component {\r\n    render() {\r\n        const [data, setData] = useState([]);\r\n        if (1 == 2) {\r\n            const [data1, setData1] = useState([]);\r\n        }\r\n        return (\r\n            <div>{data}</div>\r\n        );\r\n    }\r\n}\r\nexport default axioshooks;\r\n```\r\n\r\n",
  "closed_by": {
    "login": "gaearon",
    "id": 810438,
    "node_id": "MDQ6VXNlcjgxMDQzOA==",
    "avatar_url": "https://avatars0.githubusercontent.com/u/810438?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/gaearon",
    "html_url": "https://github.com/gaearon",
    "followers_url": "https://api.github.com/users/gaearon/followers",
    "following_url": "https://api.github.com/users/gaearon/following{/other_user}",
    "gists_url": "https://api.github.com/users/gaearon/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/gaearon/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/gaearon/subscriptions",
    "organizations_url": "https://api.github.com/users/gaearon/orgs",
    "repos_url": "https://api.github.com/users/gaearon/repos",
    "events_url": "https://api.github.com/users/gaearon/events{/privacy}",
    "received_events_url": "https://api.github.com/users/gaearon/received_events",
    "type": "User",
    "site_admin": false
  }
}
